homeassistant/components/homekit/type_fans.py

```diff
@@ -113,6 +138,18 @@ def _set_chars(self, char_values):
         if CHAR_ROTATION_SPEED in char_values:
             self.set_percentage(char_values[CHAR_ROTATION_SPEED])
 
+    def set_preset_mode(self, value, preset_mode):
+        """Set preset_mode if call came from HomeKit."""
+        _LOGGER.debug(
+            "%s: Set preset_mode %s to %d", self.entity_id, preset_mode, value
+        )
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        if value:
+            params[ATTR_PRESET_MODE] = preset_mode
+            self.async_call_service(DOMAIN, SERVICE_SET_PRESET_MODE, params)
+        else:
+            self.async_call_service(DOMAIN, SERVICE_TURN_ON, params)
```

Comment:
If value is not truthy, what are we achieving by calling turn on, reset preset mode ?

Reply:
Since there is no unset preset mode, we are accomplishing this by turning on the fan to its default on state which in turn makes the percentage slider active instead of the preset switches.